src/lazyload.intersectionObserver.js

```diff
@@ -33,7 +33,9 @@ export const observeElements = (observer, elements) => {
 export const updateObserver = (instance, elementsToObserve) => {
     resetObserver(instance._observer);
     observeElements(instance._observer, elementsToObserve);
-    processObserverEntries(instance, instance._observer.takeRecords());
+    if (typeof instance._observer.takeRecords === 'function') {
```

Comment:
I would move this guard inside the file `lazyload.environment.js`, then import it here.
How about the following line of code?

```js
export const supportsTakeRecords = supportsIntersectionObserver && "takeRecords" in IntersectionObserver.prototype;
```